### ISSUE TITLE
[question] z-wave network and range extenders

### ISSUE THREAD

**Issue by bushvin** [zwave-js/zwavejs2mqtt#1193]
I recently moved all my devices from a defective z-wave controller to a new one (gen 7 silabs SLUSB700A UZB-7).
I added 2 [range extenders](https://aeotec.com/z-wave-repeater/), but I have the impression they are not working, or not relaying the signal where it needs to go.
Unfortunately for me, the signal test is sent back to the controller as a **Power Level Report**, which I have the impression is not (yet) supported by zwavejs. So I can't tell if coverage is adequate.

When I move my Raspi zwave controller to the area with problems, all devices report as normal. Should I leave my controller where it is so all devices can try to figure out their way? Does z-wave *heal* itself and re-establish routes when it is able to connect to the controller? Will I loose control if I move my controller back to the place I want it to be, as it was before I moved it to this area?

Is there another way for me to test the range extenders?
When *interviewing* and *healing* them, there are no errors, or at least there don't seem to be any errors in the log.

Where would I start troubleshooting this?

**Comment by robertsLando** [comment]
Did you try to run a network healing? That should update return routes on every device. I dunno if you should also setup some associations in order to make extender to work @AlCalzone may help with this

**Comment by AlCalzone** [comment]
The latest version (4.2.0, driver 7.4.0) has an improved healing strategy and also makes sure that the network map is up to date. I'd start by looking at a log of the healing process in that version.

**Comment by bushvin** [comment]
upgraded to 4.2.0
I seem to be affected by this [bug](https://github.com/zwave-js/node-zwave-js/issues/2600)
I had to *re-interview* all the battery-powered nodes and manually wake them up (to speed up the process)

When launching *Heal network* all battery-powered devices were *reset* to *asleep* and *never* active.
I'll let it run for a while, waking up all nodes manually to see what happens...

**Comment by andyboeh** [comment]
I have an Aeotec Range Extender 6 and it works flawlessly without any configuration. My device has a LED that blinks whenever data is being relayed, so it's easy to tell if it's working. I cannot remember whether it worked before healing the network.

**Comment by bushvin** [comment]
After fixing the earlier mentioned bug (thanks again, @robertsLando and @AlCalzone ), I was able to proceed:

I looked for a spot where I expected to have the best coverage everywhere, and proceeded to perform a network heal. I had to wait until my battery-powered devices woke up to finish the heal, so this might have gone faster if I first had reset the sleep time to 5 minutes or so. Waking them up manually would yield varying results, and having all devices take care of it automagically worked best, apparently. Z-Wave is a bit like Noivell networks: have patience, give it a chance to fix itself.

So with a successful heal, I proceeded to move the controller to a spot closer to where it would end up (roughly halfway). I reasoned that since all my nodes already had multiple paths, they would be able to reconnect and improve any routing using updated neighbors.
I performed another network heal, and again I waited for a little over an hour for it to finish normally. Result: success! All nodes could connect to the controller. (And vice versa)

Next step: put the controller in the intended location followed by another network heal and another hour of waiting. 
Lo and behold: I have a fully functional z-wave network with each node having multiple routes to the controller!

I don't know if it would have worked if I omitted the 2nd heal, it might have, if all routes somehow ended up with the Controller.

Anyway, thank you all for chipping in! I hope this may help anyone at some point.
